### PR TITLE
Lg activity generator

### PR DESCRIPTION
**Pull request by feich-ms**
LG activity generator

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        public const string Separtor = "Separtor";
+
+        public const string AttachmentLayoutType = "AttachmentLayoutType";
+
+        public LGActivityGenerator(TemplateEngine templateEngine)
+        {
+            this.templateEngine = templateEngine;
+        }
+
+        public Activity Generate(Dictionary<string, object> options, object scope)
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs]: Looks like, making this options a free Dict is not a good idea, for better clarification and for less confusion.  I think we can make this strong typed. 
Perhaps:
class ActivityGenerationConfig {
public TextSpeakTemplateId;
public TextSpeakSeperator;
public List<AttachmentGenerationConfig> Attachments;
public AttachmentLayout Layout;
}

public class AttachmentGenerationConfig {
  public AttachementTemplateId;
  public IsAdaptiveCard (y/n);
}

File: libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                "Separtor": "||", // optional, default "||".
+                "Attachments": [
+                    { "AdaptiveCardTemplateId": "B" },
+                    { "CardTemplateId": "C" },
+                    { "CardTemplateId": "D" },
+                    { "AdaptiveCardTemplateId": "E" }], // optional if TextTemplateId is not null
+                "AttachmentLayoutType": "carousel" // optional, only used for Attachments.Count > 1, default is "carousel", optional value is "list"
+            }
+            */
+            var activity = new Activity();
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs:42]: Should we include turn context to generate Activity? how this gone be used in real conversation?

File: libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+            if (options.ContainsKey(TextTemplateId))
+            {
+                var templateId = options[TextTemplateId].ToString();
+                var separtor = options.ContainsKey(Separtor) ? options[Separtor].ToString() : "||";
+                activity = GenerateActivity(templateId, scope, separtor);
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs]: 1. Those top level tests (options.ContainsKey) makes your logic  less straghtforward
2. acitity = GenerateActivity, don't seem to align with later activity.Attachments

I would prefer

var activity = new Activity();
var TextAndSpeak = ..


(activity.Text, activity.Speak) =  GenTextAndSpeak(options.TextSpeakTemplateId, options.TextSpeakSeperator, data);
activity.Attachments = GenAttachments(options.Attachments, options.Attachment, data)
activity.AttachementLayout = options.AttachmentLayout;


return activity;

File: libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs (hunk 1 above)
Comment (feich-ms) [reply libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs]: fixed

File: libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs (hunk 3 above)
Comment (feich-ms) [reply libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs]: fixed

**Pull request closed without merging**